Extension/src/LanguageServer/Providers/callHierarchyProvider.ts

```diff
@@ -0,0 +1,199 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as vscode from 'vscode';
+import { DefaultClient } from '../client';
+import { processDelayedDidOpen } from '../extension';
+import { Position, Range, RequestType, TextDocumentIdentifier } from 'vscode-languageclient';
+import { makeVscodeRange } from '../utils';
+
+interface CallHierarchyItem {
+    /**
+     * The name of this item or symbol.
+     */
+    name: string;
+
+    /**
+     * The kind of this item.
+     */
+    kind: vscode.SymbolKind;
+
+    /**
+     * More detail for this item, e.g. the scope or class of a function.
+     */
+    detail: string;
+
+    /**
+     * The resource identifier of this item.
+     */
+    uri: string;
+
+    /**
+     * The range enclosing this symbol not including leading/trailing whitespace but everything else, e.g. comments and code.
+     */
+    range: Range;
+
+    /**
+     * The range that should be selected and revealed when this symbol is being picked, e.g. the name of a function.
+     * Must be contained by the `CallHierarchyItem.range`.
+     */
+    selectionRange: Range;
+}
+
+interface CallHierarchyParams {
+    textDocument: TextDocumentIdentifier;
+    position: Position;
+}
+
+interface CallHierarchyItemResult {
+    item?: CallHierarchyItem;
+
+    /**
+     * If a request is cancelled, `succeeded` will be undefined.
+     */
+    succeeded: boolean;

```

Comment:
The comment suggests it should be `succeeded?: boolean`?

Reply:
This is not defined as optional on the native side. But for the case when the request was cancelled. So it is not intended as `succeeded?`

Reply:
Comment has been updated to clarify condition.